lib/electric_slide/call_queue.rb

```diff
@@ -164,7 +164,7 @@ def add_agent(agent)
       abort ArgumentError.new("#add_agent called with nil object") if agent.nil?
       abort DuplicateAgentError.new("Agent is already in the queue") if get_agent(agent.id)
 
-      agent.queue = self
```

Comment:
No, that is still a problem, but the agent will probably never be re-used, and would be cleaned up when the reference from the queue is removed at the time it dies.

The difference is that the `current_actor` is a reference to the Queue actor, while `self` is a reference to the state the actor modifies, without being wrapped in the queued method dispatch of the actor.